src/gravel/controllers/nodes.py

```diff
@@ -160,13 +169,22 @@ class NodeMgr:
     _connmgr: ConnMgr
     _incoming_task: asyncio.Task
     _shutting_down: bool
-    _state: Optional[NodeStateModel]
+    _state: NodeStateModel
     _manifest: Optional[ManifestModel]
     _token: Optional[str]
 
     def __init__(self):
-        self._connmgr = ConnMgr()
         self._shutting_down = False
+        self._connmgr = ConnMgr()
+        self._manifest = None
+        self._token = None
+
+        self._init_node()
+        assert self._state
+        if self._state.stage == NodeStageEnum.READY:
```

Comment:
if the state isn't ready at this point, the NodeMgr is never started. Should this wait on an event to change the node state?

Shouldn't this also use the `self.ready` property?

Reply:
this code is somewhat obsoleted by a following commit at some stage of the patchset. Sorry :\